### ISSUE TITLE
Check symbol type during symbol table traversal

### ISSUE THREAD

**Issue by dagurval** [Bitcoin-com/cashscript#57]
```
  reason:
   'mandatory-script-verify-flag-failed (Script evaluated without error but finished with a false/empty top stack element) (code 16)',
  meep:
   'meep debug --tx=02000000017afcb5363f702752dfed9db69edadebc37da8f0c48789b5c3818cc0973b23a9d00000000fd960141fdcd5b5928a25bb7ea18b32368f1889bc2a847503a5c9964ac3d0ea999fa9799ce5b7c15dd675f9ea97a456661268620dcf82c87104381141f1fea1090216f1d4121021ed9d1795a0c4a10fabb9f4678c70015b6d4ee7a39fa2d9e5eec3b6465d6eeb04ce602000000df63e953602fb4c74a706c1d29f3f2b3aa9d5398585e8d5593b575da48c65d7218606b350cd8bf565266bc352f0caddcf01e8fa789dd8a15386327cf8cabe1987afcb5363f702752dfed9db69edadebc37da8f0c48789b5c3818cc0973b23a9d0000000049144f0b71a089a97138f977982838eb9d6c771a008378820128947f7701207f7502beef02f00d8702f00d02f00d879b69547a547a6e7c828c7f75567aa87bbbad02220258807b00aa871027000000000000feffffff64002f472744e433e490aaf46c6ed10d6a4aadd7b3a1e4c935106d64e5f84b59a87109004100000049144f0b71a089a97138f977982838eb9d6c771a008378820128947f7701207f7502beef02f00d8702f00d02f00d879b69547a547a6e7c828c7f75567aa87bbbad02220258807b00aa87feffffff02220200000000000017a9144f0b71a089a97138f977982838eb9d6c771a008387e32200000000000017a91443e0c48028dc921806ea654f15ce9b8ef57a07e287a8710900 --idx=0 --amt=10000 --pkscript=a91443e0c48028dc921806ea654f15ce9b8ef57a07e287' }
```

Code https://github.com/dagurval/cashscript/commit/f9dde2de6448ae543095e2b229d54d22a2ff30d1

**Comment by dagurval** [comment]
![Skjermbilde 2020-01-22 kl  09 18 19](https://user-images.githubusercontent.com/92707/72877041-40816800-3cf8-11ea-9065-cf42b94c4d60.png)

**Comment by rkalis** [comment]
Oh wow yeah, similar problem as the other bug. 

So the outputs need to be instantiated with the `new` keyword (`new OutputP2PKH(amount, a)`). If you forget the `new` keyword the variable it should not compile. But due to this bug, it does compile, but it compiles incorrectly.

There are two ways I can solve this right now. The first is to allow both `new Output...` and just `Output...` without the new keyword, and make sure they both compile correctly. The second is to add in the check to make sure the contract does not compile if the `new` keyword is omitted.

The second solution is the way it was supposed to be right now, but the first solution might be good as well. @dagurval what do you think of these two solutions?

**Comment by dagurval** [comment]
I'd go for only allow `new Output`, less to maintain :-).

**Comment by rkalis** [comment]
That makes sense.

**Comment by rkalis** [comment]
This is released in v0.3.2
